<?xml version="1.0"?>

<launch>
    <!--roslaunch apriltag_ros continuous_detection.launch-->
    <arg name="launch_prefix" default="" /> <!-- set to value="gdbserver localhost:10000" for remote debugging -->
    <arg name="node_namespace" default="apriltag_ros_continuous_node" />
    <arg name="camera_name" default="/camera_back" />
    <arg name="camera_frame" default="/camera_back" />
    <arg name="image_topic" default="/image_raw" />
    <rosparam command="load" file="$(find simulations)/config/settings.yaml" ns="$(arg node_namespace)" />
    <rosparam command="load" file="$(find simulations)/config/tags.yaml" ns="$(arg node_namespace)" />

    <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="$(arg node_namespace)" clear_params="true"  launch-prefix="$(arg launch_prefix)" output="log">
        <remap from="image_rect" to="$(arg camera_name)/$(arg image_topic)" />
        <remap from="camera_info" to="$(arg camera_name)/camera_info" />

        <param name="camera_frame" type="str" value="$(arg camera_frame)" />
        <param name="publish_tag_detections_image" type="bool" value="true" />      <!-- default: false -->
    </node>


    <node pkg="tf" type="static_transform_publisher" name="apriltag_to_odom" 
        args="-.33 0 0.13 0.5 0.5 -0.5 -0.5 /agv_car/base_link $(arg camera_frame)  100"/> 
        <!-- args="-.33 0 0.13 0.5 0.5 -0.5 -0.5 /agv_car/base_footprint $(arg camera_frame)  100"/>  -->
        <!-- args="-.5 0 0.2 0.5 0.5 -0.5 -0.5 /agv_car/base_footprint $(arg camera_frame)  100"/>  -->

    <!-- <node name="tag_detections_view" pkg="image_view" type="image_view" respawn="false" output="screen">
        <remap from="image" to="/tag_detections_image" />
    </node> -->
</launch>	
